packages/sdk-core/src/common/IdGenerator.ts

```diff
@@ -0,0 +1,17 @@
+import { pseudoRandomBytes } from 'crypto';
+export class IdGenerator {
+    public static uuid() {
+        const bytes = pseudoRandomBytes(16);
+        return (
+            bytes.slice(0, 4).toString('hex') +
+            '-' +
+            bytes.slice(4, 6).toString('hex') +
+            '-' +
+            bytes.slice(6, 8).toString('hex') +
+            '-' +
+            bytes.slice(8, 10).toString('hex') +
+            '-' +
+            bytes.slice(10, 16).toString('hex')
+        );
+    }
```

Comment:
This is the copy paste solution for uuid generation that we're using in node/javascript since I joined. The documentation doesn't say if this has to be v4 UUID or not. I only see this:
```
16 bytes of randomness in human readable UUID format server will reject request if uuid is already found
```

in the uuid definition.

Reply:
I believe we should be doing things according to spec: 
https://www.ietf.org/rfc/rfc4122.txt
https://github.com/uuidjs/uuid/blob/main/src/v4.js
You should also set the `clock_seq_hi_and_reserved` values.

Reply:
ok, let's treat this as a good-enough solution